lms/djangoapps/support/static/support/jsx/entitlements/data/api/client.js

```diff
@@ -0,0 +1,53 @@
+import 'whatwg-fetch';
+import Cookies from 'js-cookie';
+
+import { entitlementList } from './endpoints';
+
+const headers = {
+  Accept: 'application/json',
+  'Content-Type': 'application/json',
+  'X-CSRFToken': Cookies.get('csrftoken'),
+}
+
+export function requestEntitlements({usernameOrEmail}) {
+  return fetch(
+    `${entitlementList}/${usernameOrEmail}`, {
+      credentials: 'same-origin',
+      method: 'get'
+    },
+  );
+}
+
+export function createEntitlement({courseUuid, user, mode, reason, comments}) {
```

Comment:
Nit: for consistency it might be nice to have the ordering of parameters in the `PUT` and `POST` methods be fairly the same, i.e `{user, reason, courseUuid, mode, comments}` here to mirror below or `{entitlementUuid, email, reason, comments}` below to mirror what is here.